### DISCUSSION TITLE
LessonNine NFT at a bargain

### DISCUSSION THREAD

**Discussion by guelowrd** [Cyfrin/foundry-full-course-cu#115] [Q&A]
Hey folks,

Just finished lesson 9 -- quite a chunky one! -- and I'm now looking at claiming my extraordinary reward aka LessonNine NFT on Arbitrum.
I don't really want to spoil everybody so won't ask directly for an answer here, but perhaps just a hint... I know a way how I could claim it, but that would involve deploying a contract using R***X for instance, and as I am extremely *cheapo* I'd like to avoid that.

SO my questions is: can you confirm there is a way for me to claim what I'm due 8-) for a minimal amount (i.e. the usual gas fee, but no more)? 

And by the way, I'm taking this opportunity to say a big thanks to Patrick and all the people who helped create this course -- this is simply amazing, and I'm enjoying every second (ok, almost every second). Congrats, and cheers to you all.

Gaylord

**Comment by alymurtazamemon** [comment]
@guelowrd I think NO

**Comment by PatrickAlphaC** [comment]
There is a clever way you can mint it without deploying a contract... but it is much harder.

**Reply by guelowrd** [reply]
Hmm interesting... Using  a few cast calls from CLI, I should be able to retrieve info about the block status and then call LessonNine contract? Or is it too naive...

**Reply by f3d0ss** [reply]
> Hmm interesting... Using a few cast calls from CLI, I should be able to retrieve info about the block status and then call LessonNine contract? Or is it too naive...

I think on Sepolia this will not work because your transaction will be included in the next block and the `block.prevrandao` and `block.timestamp` will be different.

BTW I didn't yet found a solution without deploying a new contract

**Reply by tbrannt** [reply]
just solved it on sepolia without deploying a smart contract
